jsx/Help.js

```diff
@@ -0,0 +1,151 @@
+  };
+
+  useEffect(() => {
+    props.content && setContent(props.content);
+  }, [props.content]);
+
+  useEffect(() => {
+    !props.content && fetch(
+      loris.BaseURL
+      + '/help_editor/ajax/help.php?'
+      + new URLSearchParams(getParams),
+      {method: 'GET'},
+    ).then((response) => {
+      if (!response.ok) {
+        console.error(response.status);
+        return;
+      }
+      response.json().then(
+        (data) => {
+          if (!isMounted) return;
+          setFormat(data.format);
+
+          // Render Markdown in wrap div.
+          // If help content is from Markdown helpfile.
+          if (format === 'markdown') {
+            setContent(<RMarkdown content={data.content}/>);
+          } else {
+            // If help content is from DB.
+            setContent(
+              <>
+                {data.topic && <h3>{data.topic}</h3>}
+                <div><RMarkdown content={data.content}/></div>
+                {data.updated && <><hr/>Last updated: {data.updated}</>}
+              </>
+            );
+          }
+        }
+      );
+    }).catch((error) => {
+      // Network error
+      console.error(error);
+    });
+
+    return () => setIsMounted(false);
+  }, []);
+
+  let editBtn = null;
+  // If help content comes from DB `help` table and can
+  // be edited online.
+  if (
+    loris.userHasPermission('context_help')
+    && format !== 'markdown'
+    && !props.content
+  ) {
+    editBtn = <button
+      className="btn btn-default"
+      style={{
+        float: 'right',
+        margin: '10px',
+      }}
+      onClick={(e) => {
+        e.preventDefault();
+        window.open(
+          loris.BaseURL
+          + '/help_editor/edit_help_content/?section='
+          + getParams.testName
+          + '&subsection='
+          + getParams.subtest,
+          '_self'
+        );
+      }}
+    >Edit</button>;
+  }
+
+  /**
+   * Renders the React component.
+   *
+   * @return {JSX} - React markup for the component
+   */
+  return (
+    <>
+      <button
+        type="button"
+        className={'help-button' + (isActive ? ' help-open' : '')}
+        onClick={() => setIsActive(!isActive)}
+      >
+        <img width="17" src="http://localhost/images/help.gif" />
+      </button>
+
+      <div
+        className="help-content"
+        style={{
+          display: isActive ? 'block' : 'none',
+        }}
+      >
+        <button
+          className="btn btn-default"
+          style={{
+            float: 'right',
+            margin: '10px 10px 10px 0',
+          }}
+          onClick={() => setIsActive(!isActive)}
+        >Close</button>
+        {editBtn}
+        <div className="help-wrapper">
+          {content}
+        </div>
+      </div>
+    </>
+  );
+};
+
+document.addEventListener('DOMContentLoaded', () => {
+  const helpContainers = document.getElementsByClassName('help-container');
+  for (let i = 0; i < helpContainers.length; i++) {
```

Comment:
I don't understand the purpose of this loop. Why are there multiple "help containers"? We only have one help icon.

Reply:
We do have 2. I agree this is not ideal but it is actually design this way. One for the desktop view, one for the mobile view. We only see one at a time.